src/components/footer-reservation/FooterReservation.type.ts

```diff
@@ -0,0 +1,13 @@
+export interface IFooter extends IButtonType {
+  isBookmarked: boolean
+  isReservationProduct: boolean
+  discount?: number
+  price?: number
+  buttontype: 'reservation' | 'payment' | 'cancelPayment'
```

Comment:
P3

 IButtonType extends해서 쓴 거면 IButtonType에 있는 타입은 추가로 안 써도 되지 않나요?
+코드는 위에서 아래로 읽으니까, 순서도 바꾸면 좋을 것 같아요

```suggestion
export interface IButtonType {
  buttontype: 'reservation' | 'payment' | 'cancelPayment'
}

export interface IFooter extends IButtonType {
  isBookmarked: boolean
  isReservationProduct: boolean
  discount?: number
  price?: number
  productId?: string
  onPayConfirmClick?: () => void
}
```

또는, IButtonType 인터페이스에서 buttonType만 정해준다면, type으로 빼서 관리해도 좋을 것 같아요

```suggestion
export type ButtonType =  'reservation' | 'payment' | 'cancelPayment'

export interface IFooter extends IButtonType {
  isBookmarked: boolean
  isReservationProduct: boolean
  discount?: number
  price?: number
  buttontype : ButtonType
  productId?: string
  onPayConfirmClick?: () => void
}